services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/EphemerySlotValidationService.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.services.beaconchain;
+
+import static tech.pegasys.teku.infrastructure.unsigned.UInt64.ZERO;
+
+import tech.pegasys.teku.ethereum.events.SlotEventsChannel;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.service.serviceutils.Service;
+import tech.pegasys.teku.storage.client.RecentChainData;
+
+public class EphemerySlotValidationService extends Service implements SlotEventsChannel {
+
+  private final RecentChainData recentChainData;
+
+  public EphemerySlotValidationService(final RecentChainData recentChainData) {
+    this.recentChainData = recentChainData;
+  }
+
+  @Override
+  public void onSlot(final UInt64 slot) {
+    final UInt64 currentSlot =
+        recentChainData
+            .getCurrentSlot()
+            .orElseThrow(() -> new IllegalStateException("Current slot not available"));
```

Comment:
the slot comes into the function, you can just use that..

Reply:
Not sure what you meant

Reply:
`onSlot(final UInt slot) {`
The slot is the slot we're entering... we don't need to grab it from recentChainData, it's passed in.